feat: added `xpython` and `xсontext` builtin aliases

Hammer2900 opened the pull request:
### Description

#### xcontext

This PR adds a new built-in alias `xontext`. It prints useful debug information about the current Xonsh session and environment (Python paths, Xonsh paths, pip, and key environment variables).

It is based on the `xc` alias from `xontrib-rc-awesome`. This helps users and maintainers quickly debug environment issues.

```xsh
xontext
# [Current xonsh session]
# xpython: /home/izot/Downloads/___/xonsh/.venv_f/bin/python # Python 3.12.10
# xpip: /home/izot/Downloads/___/xonsh/.venv_f/bin/python -m pip
# 
# [Current commands environment]
# xonsh: /home/izot/Downloads/___/xonsh/.venv_f/bin/xonsh
# python: /home/izot/Downloads/___/xonsh/.venv_f/bin/python # Python 3.12.10
# pip: /home/izot/Downloads/___/xonsh/.venv_f/bin/pip
# pytest: /home/izot/.local/bin/pytest

# VIRTUAL_ENV: /home/izot/Downloads/___/xonsh/.venv_f
```

Closes #5400 

#### xpython

Added `xpython` as reference to python that was used to run current session of xonsh.

anki-code reviewed (changes requested):
Hey, that's nice to have! Some points:
1. Let's rename it to `xcontext` it sounds more explicit.
2. Test the case when you have no python in $PATH (e.g. [xonsh-install](https://github.com/anki-code/xonsh-install)) and beutify the result output.
3. Check that new allias will be in https://xon.sh/aliases.html. I'm just don't remember do we need an addition action for this.
4. Please add `xpython` alias to run xonsh's python. (https://github.com/xonsh/xonsh/pull/5989#issuecomment-3654882563)

jnoortheen commented:
> @jnoortheen we have `xpip` but I have some cases in [AppImage](https://xon.sh/appimage.html), in [xxh-shell-xonsh](https://github.com/xxh/xxh-shell-xonsh) and in [xonsh-install](https://github.com/anki-code/xonsh-install) when it would be nice to call `xpython` e.g. `xpip install lolcat && xpython -m lolcat`. What do you think about adding `xpython` here as well?

Sure, `xpython` can be added

Hammer2900 commented:
Done!

1. Renamed xontext to xcontext.
2. Added xpython alias (pointing to sys.executable).
3. Migrated to ArgParserAlias (using _args, _stdin, etc. to hide them from help).
4. Added checks for missing commands in output (prints "not found" instead of crashing).

anki-code commented:
Looks great! @Hammer2900 finally please add them to the docs - https://github.com/xonsh/xonsh/blob/main/docs/aliases.rst

anki-code commented:
Good work @Hammer2900 !
